examples/src/main/java/com/datastax/oss/driver/examples/concurrent/LimitConcurrencyCustom.java

```diff
@@ -0,0 +1,148 @@
+
+  // Create CountDownLatch that wait for completion of all pending requests
+  private static final CountDownLatch REQUEST_LATCH = new CountDownLatch(TOTAL_NUMBER_OF_INSERTS);
+
+  public static void main(String[] args) throws InterruptedException {
+
+    try (CqlSession session = new CqlSessionBuilder().build()) {
+      createSchema(session);
+      insertConcurrent(session);
+    }
+  }
+
+  private static void insertConcurrent(CqlSession session) throws InterruptedException {
+    PreparedStatement pst =
+        session.prepare(
+            insertInto("examples", "tbl_sample_kv")
+                .value("id", bindMarker("id"))
+                .value("value", bindMarker("value"))
+                .build());
+
+    // Used to track number of total inserts
+    AtomicInteger insertsCounter = new AtomicInteger();
+    // Used to track threads that were involved in a processing
+    Set<String> threads = new ConcurrentSkipListSet<>();
+
+    // Executor service with CONCURRENCY_LEVEL number of threads that states an upper limit
+    // on number of request in progress.
+    ExecutorService executor = Executors.newFixedThreadPool(CONCURRENCY_LEVEL);
+
+    // For every i we will insert a record to db
+    for (int i = 0; i < TOTAL_NUMBER_OF_INSERTS; i++) {
+      // Before submitting a request, we need to acquire 1 permit.
+      // If there is no permits available it blocks caller thread.
+      SEMAPHORE.acquire();
+      // Copy to final variable for usage in a separate thread
+      final int counter = i;
+
+      // We are running CqlSession.execute in a separate thread pool (executor)
+      CompletableFuture.supplyAsync(
+          () -> {
+            insertsCounter.incrementAndGet();
+            threads.add(Thread.currentThread().getName());
+
+            ResultSet executeRequest;
+            try {
+              executeRequest =
+                  session.execute(
+                      pst.bind().setUuid("id", UUID.randomUUID()).setInt("value", counter));
+            } finally {
+              // Signal that processing of this request finishes
+              REQUEST_LATCH.countDown();
+              // Once the request is executed, we release 1 permit.
+              // By doing so we allow caller thread to submit another async request.
+              SEMAPHORE.release();
+            }
+
+            return executeRequest;
+          },
+          // Here the separate thread pool is passed as the argument
+          executor);
+    }
+    // Await for execution of TOTAL_NUMBER_OF_INSERTS
+    REQUEST_LATCH.await();
+
+    System.out.println(
+        String.format(
+            "Finished executing %s queries with a concurrency level of %s.",
+            insertsCounter.get(), threads.size()));
+    // Shutdown executor to free resources
+    executor.shutdown();
+  }
+
+  private static void createSchema(CqlSession session) {
+    session.execute(
```

Comment:
If you have a multi-node cluster, its possible that the schema queries will timeout after 2s.  Maybe we should add a `schema` profile to application.conf and have all our examples use it.

Reply:
Good idea, I think it will be good to do it in a separate PR.
I've created JIRA ticket to capture that:
https://datastax-oss.atlassian.net/browse/JAVA-2318